### DISCUSSION TITLE
Styled widgets (StyledText, StyledIcon...) don't work like Box widgets

### DISCUSSION THREAD

**Discussion by Maksimka101** [btwld/mix#319] [General]
It seems strange to me that the `StyledText` and `FlexBox` widgets accept `Style`, but only `FlexBox` processes `$box` attributes, while `StyledText` quietly ignores them. Firstly, it's very unpleasant to pass an attribute that isn't applied. Secondly, it's inconvenient that to create a decorator for `Row`, you can use `HBox`, but to create a decorator for `Text`, you need to use `Box + StyledText`. And in general, it's strange that modifiers apply to everything, context variants also always work, but `$box` works almost always. Why not follow the idea through and create `TextBox`, `ImageBox`, and `IconBox` so that the `$box` attribute always works?

And secondly, most importantly, I did not notice any mention in the documentation that `Styled` widgets do not work with the `$box` attribute

So, my question is: should absolutely any widget from the mix package support the `$box` attribute?

**Comment by leoafarias** [comment]
@Maksimka101 Thanks for the feedback; we have a work-in-progress PR to allow modifiers per Spec.

Regarding your question, the plan is to try to have a 1-to-1 mapping of the utilities. I believe FlexBox is the only exception to this. 

If you have a $box and $text in the same style, you can pass them to the widget Box and StyledText, and their attributes will be resolved.

However, StyledText and StyledIcon inherit $text and $icon attributes passed straight to Box.

**Comment by tilucasoli** [comment]
Recently, we merged a PR (#517) that introduced a utility specific for `FlexBox`. Now, you can write your styles, such as: 

```dart
$flexBox.chain
  ..alignment.center()
  ..padding(10)
  ..margin(10)
  ..flex.mainAxisAlignment.center()
  ..crossAxisAlignment.center()
```
